### PR TITLE
encode urls as uris so that URI.create is ok

### PR DESCRIPTION
**Pull request by bergman**
# Hey, I just made a Pull Request!

## Description
encode urls in a more compatible way

## Motivation and Context
apollo client seems to use URI.create before passing to underlying
client and URI.create throws on unencoded characters like [ and ]

## Have you tested this? If so, how?
I have included unit tests.

## Checklist for PR author(s)
- [ ] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)
- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

**Comment by honnix** [comment]
codecov is unhappy about delta changes. :) Want to spend time to cover more?

**Comment by bergman** [comment]
I added a few more tests that weird characters are encoded correctly. Let's see what coverage is now.

File: styx-client/src/test/java/com/spotify/styx/client/StyxApolloClientTest.java [hunk 1]
```diff
@@ -340,16 +360,16 @@ public void testTokenSuccess() {
...
-  public void testTokenFailure() throws Exception {
+  public void testTokenFailure() throws IOException, java.security.GeneralSecurityException {
     final IOException rootCause = new IOException("netsplit!");
     when(auth.getToken(any())).thenThrow(rootCause);
     final StyxApolloClient styx = new StyxApolloClient(client, CLIENT_HOST, auth);
     final CompletableFuture<Void> f = styx.triggerWorkflowInstance("foo", "bar", "baz")
         .toCompletableFuture();
     try {
-      f.get();
+      f.join();
```
Comment (honnix) [inline styx-client/src/test/java/com/spotify/styx/client/StyxApolloClientTest.java]: Why the change here?

File: styx-client/src/test/java/com/spotify/styx/client/StyxApolloClientTest.java [hunk 2]
```diff
@@ -340,16 +360,16 @@ public void testTokenSuccess() {
   }
 
   @Test
-  public void testTokenFailure() throws Exception {
+  public void testTokenFailure() throws IOException, java.security.GeneralSecurityException {
```
Comment (honnix) [inline styx-client/src/test/java/com/spotify/styx/client/StyxApolloClientTest.java]: We don't need to be so specific I think for test cases, but fine.

File: styx-client/src/test/java/com/spotify/styx/client/StyxApolloClientTest.java (hunk 1 above)
Comment (bergman) [reply styx-client/src/test/java/com/spotify/styx/client/StyxApolloClientTest.java]: unchecked exception, nicer to handle than `ExecutionException` etc

File: styx-client/src/test/java/com/spotify/styx/client/StyxApolloClientTest.java (hunk 2 above)
Comment (bergman) [reply styx-client/src/test/java/com/spotify/styx/client/StyxApolloClientTest.java]: Agreed, will revert

File: styx-client/src/test/java/com/spotify/styx/client/StyxApolloClientTest.java (hunk 1 above)
Comment (bergman) [reply styx-client/src/test/java/com/spotify/styx/client/StyxApolloClientTest.java]: You're right, in this context it doesn't help much. I was too eager :)

**Comment by bergman** [comment]
@honnix How about going for the okhttp PR instead of this one? #584

**Comment by honnix** [comment]
> How about going for the okhttp PR instead of this one? #584
Sounds greater!

**Comment by bergman** [comment]
Closing since #584 was merged
